unit/goto-programs/goto_program_function_call.cpp

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************\
+
+  Module: Unit tests for goto_program::validate
+
+  Author: Diffblue Ltd.
+
+ \*******************************************************************/
+
+#include <goto-programs/goto_function.h>
+#include <testing-utils/catch.hpp>
+#include <util/arith_tools.h>
+
+SCENARIO(
+  "Validation of well-formed function call codes",
+  "[core][goto-programs][validate]")
+{
+  GIVEN("A program with one function call")
+  {
+    symbol_tablet symbol_table;
+    const typet type1 = signedbv_typet(32);
```

Comment:
I'd prefer using `const signedbv_typet type1` here. (though ultimately we'll have to assign this type to a `typet` anyways when assigining to the `type` field of a `symbolt`).